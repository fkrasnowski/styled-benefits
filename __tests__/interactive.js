import { interactive } from '../.'
import styled from '@emotion/styled'
import { shallow, mount } from 'enzyme'
import React from 'react'

test('Interactive is defined', () => {
  expect(interactive).toBeDefined()
})

describe('Interactive Component;', () => {
  const Component = interactive(styled.div``)

  test('Interactive component should render correctly', () => {
    const component = shallow(<Component />)
    expect(component).toMatchSnapshot()
  })

  test('onClick handler should be called', () => {
    const clickFn = jest.fn()
    const component = mount(<Component onClick={clickFn} />)
    component.props().onClick()

    expect(clickFn).toHaveBeenCalledTimes(1)
    component.unmount()
  })

  test.todo('ref should reference to the DOM element')
})

test('Should access chained props', () => {
  expect(
    interactive(styled.div`
      ${props => {
        props.it[0].style().get()
        return ''
      }}
    `)
  ).not.toThrow()
})
