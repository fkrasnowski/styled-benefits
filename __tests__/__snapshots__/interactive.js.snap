// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interactive Component; Interactive component should render correctly 1`] = `
<SelectorProvider
  Element={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "__emotion_base": "div",
      "__emotion_forwardProp": undefined,
      "__emotion_real": [Circular],
      "__emotion_styles": Array [
        "",
      ],
      "defaultProps": undefined,
      "displayName": "Styled(div)",
      "render": [Function],
      "withComponent": [Function],
    }
  }
  delays={Object {}}
  keyframesProps={Array []}
  props={Object {}}
/>
`;
